api/src/main/java/org/openmrs/module/fhir/api/util/FHIRImagingStudyUtil.java

```diff
@@ -0,0 +1,196 @@
+			Patient patient = FHIRRESTfulGenericClient.readPatientById(serverBase, uuid);
+			List<String> errors = new ArrayList<String>();
+			omrsPatient = FHIRPatientUtil.generateOmrsPatient(patient, errors);
+			if (errors.isEmpty()) {
+				// Save the retrieved Patient
+				omrsPatient = Context.getPatientService().savePatient(omrsPatient);
+				return omrsPatient;
+			} else {
+				StringBuilder errorMessage = new StringBuilder(
+						"The request cannot be processed due to the following issues\n");
+				for (int i = 0; i < errors.size(); i++) {
+					errorMessage.append(i + 1).append(" : ").append(errors.get(i)).append("\n");
+				}
+				throw new UnprocessableEntityException(errorMessage.toString());
+			}
+		} else {
+			return omrsPatient;
+		}
+	}
+
+	public static Obs generateOpenMRSSeriesObs(ImagingStudy.Series series, org.openmrs.Patient omrsPatient) {
+		Concept conceptSeries = FHIRUtils.getImagingStudySeriesConcept();
+		// Set `dateTime` as Obs DateTime
+		Date dateTime = series.getDateTime();// Will not be null
+		// Create a Obs for store Series
+		Obs omrsSeries = new Obs(omrsPatient, conceptSeries, dateTime, null);
+
+		String value = "";
+		// Set `number` as Obs Value (Text)
+		value += "number:".concat(series.getNumber().toString() + ",");
+		// Set `modality` as Obs Value (Text)
+		value += "modality:".concat(series.getModality() + ",");
+		// Set `uid` as Obs uuid
+		omrsSeries.setUuid(series.getUid());
+		// Set `description` as Obs Comment
+		omrsSeries.setComment(series.getDescription());
+		// `numberOfInstance` = Number of Instance Obs
+		// Set `url` as Obs Value (Text)
+		value += "url:".concat(series.getUrl() + ",");
+
+		// Set All values
+		omrsSeries.setValueText(value);
+		// Set `instance` as Obs group
+		ObsService obsService = Context.getObsService();
+		for (ImagingStudy.SeriesInstance instance : series.getInstance()) {
+			Obs omrsInstance = generateOpenMRSInstanceObs(instance, omrsPatient);
+			omrsInstance = obsService.saveObs(omrsInstance, null);
+			omrsSeries.addGroupMember(omrsInstance);
+		}
+		return omrsSeries;
+	}
+
+	public static Obs generateOpenMRSInstanceObs(ImagingStudy.SeriesInstance instance, org.openmrs.Patient omrsPatient) {
+		Concept conceptInstance = FHIRUtils.getImagingStudySeriesInstanceConcept();
+		Obs omrsInstance = new Obs();
+		omrsInstance.setPerson(omrsPatient);
+		String value = "";
+		// Set `number` as Obs Value (Text)
+		value += "number:".concat(instance.getNumber().toString() + ",");
+		// Set `uid` as Obs uuid
+		omrsInstance.setUuid(instance.getUid());
+
+		// Set All values
+		omrsInstance.setValueText(value);
+		// Set `content` as Complex Obs
+		for (AttachmentDt attachment : instance.getContent()) {
+			int conceptId = FHIRUtils.getImagingStudySeriesInstanceContentConcept().getConceptId();
+			if (attachment.getCreation() == null) {
+				attachment.setCreation(new DateTimeDt());
+			}
+			System.out.println("1111111111111 size " + attachment.getSize());
```

Comment:
@harsha89 I tried with following code
```javascript
protected static final Log log = LogFactory.getLog(this.getClass());
```
But `this.getClass()` is not working here, bcz of ".this can't reference from a static context". How to overcome that ?